x/gamm/keeper/pool_service.go

```diff
@@ -249,18 +249,22 @@ func (k Keeper) JoinSwapExactAmountIn(
 	}
 
 	sharesOut, err := pool.JoinPool(ctx, tokensIn, pool.GetSwapFee(ctx))
-	if err != nil {
+	switch {
```

Comment:
note to reviewer: Converted cascading if/else to a switch, which is much much easier to grok